src/mapDataManipulation/mapDataManipulationUtils.ts

```diff
@@ -10,38 +10,54 @@ export function prepareRgbMappingArrays(): RgbMappingArrays {
 
 export function changeRgbMappingArraysWithFunction(
   rgbMappingArrays: RgbMappingArrays,
-  transfrormationFunction: Function,
+  transformationFunction: Function,
 ): RgbMappingArrays {
-  rgbMappingArrays.red = rgbMappingArrays.red.map(x => transfrormationFunction(x));
-  rgbMappingArrays.green = rgbMappingArrays.green.map(x => transfrormationFunction(x));
-  rgbMappingArrays.blue = rgbMappingArrays.blue.map(x => transfrormationFunction(x));
+  rgbMappingArrays.red = rgbMappingArrays.red.map(x => transformationFunction(x));
+  rgbMappingArrays.green = rgbMappingArrays.green.map(x => transformationFunction(x));
+  rgbMappingArrays.blue = rgbMappingArrays.blue.map(x => transformationFunction(x));
   return rgbMappingArrays;
 }
 
-// from one interval to another
+// from one range to another
 // f(x) = c + ((d - c) / (b - a)) * (x - a)
 // a = oldMin, b = oldMax; c = newMin, d = newMax
 // [0,255] to [0,1]: a = 0, b = 255; c = 0, d = 1
 // [0,1] to [0,255]: a = 0, b = 1; c = 0, d = 255
 
-export function changeRgbMappingArraysInterval(
+export function changeRgbMappingArrayRange(
+  rgbMappingArray: number[],
+  oldMin: number,
+  oldMax: number,
+  newMin: number,
+  newMax: number,
+): number[] {
+  const transformValueToRange = (x: number): number => {
+    let newX = newMin + ((newMax - newMin) / (oldMax - oldMin)) * (x - oldMin);
+    newX = Math.max(newX, newMin);
+    newX = Math.min(newX, newMax);
+
+    return newX;
+  };
```

Comment:
As we have discussed, I would suggest moving this function outside to avoid code duplication. 

It is also possible that you will be left with function(s) with a single statement, in which case I would suggest removing them and performing that statement directly.

Still, this is mostly a cleanup, so fftr.

Reply:
Thanks, did that. It could probably be cleaned up even further, but maybe do that in a separate MR.